File: backend/app/schemas/analyze.py
from pydantic import BaseModel, HttpUrl
from typing import List, Optional

class AnalyzeRequest(BaseModel):
    repository_url: HttpUrl

class UndocumentedFile(BaseModel):
    file: str
    issue: str
    severity: str

class Suggestion(BaseModel):
    suggestion: str

class CategorizedMetrics(BaseModel):
    readme_completeness: int
    api_documentation_coverage: int
    onboarding_quality: int
    missing_documentation_percentage: int

class RepoSummary(BaseModel):
    total_files: int
    has_readme: bool
    code_files: int
    doc_files: int
    health_score: int
    metrics: Optional[CategorizedMetrics] = None

class AnalyzeResponse(BaseModel):
    owner: str
    repo: str
    description: Optional[str] = None
    stars: int = 0
    forks: int = 0
    open_issues: int = 0
    summary: RepoSummary
    warnings: List[UndocumentedFile]
    suggestions: List[Suggestion]
    generated_docs: Optional[str] = None

File: frontend/src/types/analyze.ts
export interface CategorizedMetrics {
  readme_completeness: number;
  api_documentation_coverage: number;
  onboarding_quality: number;
  missing_documentation_percentage: number;
}

export interface RepoSummary {
  total_files: number;
  has_readme: boolean;
  code_files: number;
  doc_files: number;
  health_score: number;
  metrics?: CategorizedMetrics;
}

export interface UndocumentedFile {
  file: string;
  issue: string;
  severity: string;
}

export interface Suggestion {
  suggestion: string;
}

export interface AnalyzeResponse {
  owner: string;
  repo: string;
  description?: string;
  stars: number;
  forks: number;
  open_issues: number;
  summary: RepoSummary;
  warnings: UndocumentedFile[];
  suggestions: Suggestion[];
  generated_docs?: string;
}

export interface GitHubUser {
  id: number;
  login: string;
  name?: string | null;
  avatar_url?: string | null;
  html_url?: string | null;
}

export interface AuthSession {
  authenticated: boolean;
  user?: GitHubUser | null;
}

export interface GitHubRepository {
  id: number;
  name: string;
  full_name: string;
  private: boolean;
  html_url: string;
  description?: string | null;
  default_branch: string;
  language?: string | null;
  stargazers_count: number;
  forks_count: number;
  updated_at?: string | null;
}
